doc/tools/make_rst.py

```diff
@@ -974,85 +1005,181 @@ def make_rst_class(class_def: ClassDef, state: State, dry_run: bool, output_dir:
                 format_table(f, info)
 
             if theme_item_def.text is not None and theme_item_def.text.strip() != "":
-                f.write(rstize_text(theme_item_def.text.strip(), theme_item_def, state) + "\n\n")
+                f.write(format_text_block(theme_item_def.text.strip(), theme_item_def, state) + "\n\n")
 
             index += 1
 
     f.write(make_footer())
 
 
-def escape_rst(text: str, until_pos: int = -1) -> str:
-    # Escape \ character, otherwise it ends up as an escape character in rst
-    pos = 0
-    while True:
-        pos = text.find("\\", pos, until_pos)
-        if pos == -1:
-            break
-        text = text[:pos] + "\\\\" + text[pos + 1 :]
-        pos += 2
+def make_type(klass: str, state: State) -> str:
+    if klass.find("*") != -1:  # Pointer, ignore
+        return klass
+    link_type = klass
+    if link_type.endswith("[]"):  # Typed array, strip [] to link to contained type.
+        link_type = link_type[:-2]
+    if link_type in state.classes:
+        return ":ref:`{}<class_{}>`".format(klass, link_type)
+    print_error('{}.xml: Unresolved type "{}".'.format(state.current_class, klass), state)
```

Comment:
Since we are using Python >= 3.6, maybe consider using f-strings throughout for slightly improved readability?

```py
print_error(f'{state.current_class}.xml: Unresolved type "{klass}".', state)
```

(Especially with something like `make_link` below with all the `+` concatenation.)

Reply:
I don't mind this change, but it's probably worth doing it in a subsequent PR for all the strings.

Reply:
👍 I'm happy to do that if you are fine with the idea.

Reply:
Agree this could be good, also agree it should go in a separate PR